automation/tests-storage.sh

```diff
@@ -12,28 +33,107 @@ create_loop_devices() {
 }
 
 setup_storage() {
-    python3 tests/storage/userstorage.py setup
+    # Configure lvm to ignore udev events, otherwise some lvm tests hang.
+    mkdir -p /etc/lvm
+    cp docker/lvmlocal.conf /etc/lvm/
+
+    # Make sure we have enough loop device nodes. Using 16 devices since with 8
+    # devices we have random mount failures.
+    create_loop_devices 16
+
+    # Build vdsm.
+    ./autogen.sh --system
+    make
+
+    # Setup user stoage during the tests.
+    make storage
 }
 
 teardown_storage() {
-    python3 tests/storage/userstorage.py teardown \
+    make clean-storage \
         || echo "WARNING: Ingoring error while tearing down user storage"
 }
 
-# Configure lvm to ignore udev events, otherwise some lvm tests hang.
-mkdir -p /etc/lvm
-cp docker/lvmlocal.conf /etc/lvm/
+# Process command-line arguments
+while test $# -gt 0
+    do
+        case $1 in
 
-# Make sure we have enough loop device nodes. Using 16 devices since with 8
-# devices we have random mount failures.
-create_loop_devices 16
+            root )
 
-# Build vdsm.
-./autogen.sh --system
-make
+                mode=root
+                ;;
+            
+            vdsm )
+
+                mode=vdsm
+                ;;
+
+            -u=* | --user=* )
+
+                user="${1#*=}"
+                if [ ! -z ${mode+x} ]; then
+                    mode=vdsm
+                fi
+                ;;
+            
+            -u | --user )
+
+                shift
+                if [ -z $1 ]; then
+                    echo "error: user expected after --user"
+                    exit 1
+                fi
+                user="$1"
+                if [ ! -z ${mode+x} ]; then
+                    mode=vdsm
+                fi
+                ;;
+            
+            * )
+
+                echo "Unknown option: $1"
+                print_help
+                exit 1
+            ;;
+
+        esac
+        shift
+done
```

Comment:
Do we really need 35 lines to set the mode?

Why not one line?

    mode=${$1:?mode required}

Or:

    mode=$1
    if [ "$mode" != "vdsm" -a "$mode" != "root" ]; then
        show usage...
    fi

Reply:
Simplifying the CLI as in https://github.com/oVirt/vdsm/pull/276#discussion_r915292609 will simplify the code to parse arguments, and make it more similar to what you wrote.

In this approach, I just didn't want to assume that the mode is first argument and user second, so I parsed them in a loop.